### PR TITLE
Upgrade to Composer 2

### PR DESCRIPTION
**Pull request by xendk**
No changes to composer.{json|lock} needed, but install Composer 2.0.3 in the container.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,9 +4,13 @@ RUN install_clean php-cli php-curl php-xml php-mbstring git unzip
...
-RUN curl -sS https://getcomposer.org/installer | php && \
-    mv composer.phar /usr/local/bin/composer && \
-    cd /bug-yield && composer install && \
-    crontab /bug-yield/docker/crontab
+RUN curl -o /tmp/composer-installer https://getcomposer.org/installer && \
+        curl -o /tmp/composer-installer.sig https://composer.github.io/installer.sig &&  \
+        php -r "if (hash('SHA384', file_get_contents('/tmp/composer-installer')) !== trim(file_get_contents('/tmp/composer-installer.sig'))) { unlink('/tmp/composer-installer'); echo 'Invalid installer' . PHP_EOL; exit(1); }" && \
+        php /tmp/composer-installer --version=2.0.3 --filename=composer --install-dir=/usr/local/bin && \
+        php -r "unlink('/tmp/composer-installer');" && \
+        php -r "unlink('/tmp/composer-installer.sig');" && \
```
Comment (kasperg) [inline Dockerfile:12]: Can we simplify this with multistage builds using the official Docker image?

By copying from `composer:2` we should be able to get the latest version without running into release 3 at some point. IMO Composer has a good track recording regarding stability.

**Review by kasperg — APPROVED** [review]
:+1: LGTM. I have a single question you may want to look into.

Merge as you please :smile:.

File: Dockerfile (hunk 1 above)
Comment (xendk) [reply Dockerfile:12]: Of course.

Composer has `2.0.4` tags, but I'm all for auto-updating.

Comment (arnested) [reply Dockerfile]: You could setup @dependabot to also create PR's for updates to the Dockerfile: https://github.com/spejder/msml/blob/master/.github/dependabot.yml#L12-L16

Comment (xendk) [reply Dockerfile]: @arnested 
Doesn't it do that by default? It's enabled for the repo and creates PRs for composer packages.

Comment (arnested) [reply Dockerfile]: It's only enabled for composer: https://github.com/reload/bug-yield/blob/master/.github/dependabot.yml

Comment (xendk) [reply Dockerfile]: Oh, I forgot the config file is in `.github`.. .I'll add docker.

**Pull request merged**